storage/src/main/java/org/apache/kafka/server/log/internals/LeaderEpochCheckpoint.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.internals;
+
+import java.util.List;
+
+public interface LeaderEpochCheckpoint {
```

Comment:
Looks like this was in the checkpoints package previously. Would it make sense to keep that distinction here? It could be `org.apache.kafka.server.checkpoint.internals` or something along those lines.

Reply:
That said, should this be in the `storage` module at all?

Reply:
It makes sense to me that it should be. But perhaps the package structure for this module should be `org.apache.kafka.server.storage.internals.*` versus the one we've been using for the log layer. Maybe we can use this approach for the checkpoint code and move the log layer all at once at the end.

Reply:
Thinking about this some more, to remain consistent with other packages like `raft`, `metadata` and others, we could skip the `server` and just have `org.apache.kafka.storage.internals.*` for the non api classes.

Reply:
I originally also preferred having `server` in the package name, but on further thought, there are advantages in basically having the module name as the package name after `kafka`. It avoids accidental split packages across modules, which can cause problems if we ever decide to use Java 9 modules.

If we agree on that approach, then the module names would be `org.apache.kafka.storage.internals.checkpoint` and `org.apache.kafka.storage.internals.log` (one annoying aspect of using `log` is that GitHub filters it from their code navigation, but we probably have to live with that for now).

@junrao what are your thoughts?

Reply:
Yes, the proposed package names sound good to me.

Reply:
I moved checkpoint files to `org.apache.kafka.storage.internals.checkpoint` with the latest commits as mentioned earlier. We can have a separate followup PR for moving existing `log.internals`.

Reply:
Filed [KAFKA-14688](https://issues.apache.org/jira/browse/KAFKA-14688) and it is addressed with https://github.com/apache/kafka/pull/13213